content/blog/2016/2016-04-07-pipeline-for-runs-on-hardware.adoc

```diff
@@ -0,0 +1,258 @@
+---
+layout: post
+title: Automating test runs on hardware with Pipeline-as-Code. Part 1
+tags:
+- jenkins
+- jenkins2
+- pipeline
+- embedded
+author: oleg_nenashev
+---
+
+In addition to Jenkins development, during last 8 years I've been involved into continuous integration for hardware and embedded projects.
+At JUC2015/London 
+link:https://www.cloudbees.com/jenkins/juc-2015/presentations/JUC-2015-Europe-Jenkins-Based-CI-for-Nenashev.pdf[I have conducted a talk] about common automation challenges in the area.
+
+In this blog post I would like to concentrate on link:https://jenkins.io/doc/pipeline/[Pipeline] (fka Workflow), which is a new automation-as-code ecosystem for Jenkins.
+It will be proposed by default in the incoming link:/2.0/[Jenkins 2.0 release].
+
+First time I've tried Pipeline two and half years ago, and unfortunately it didn't work for my cases at all. 
+I was very disappointed and tried it only one year later, At this point the plugin became much more stable and useful. 
+It also attracted much more contributors, and it started evolving rapidly and finally became a huge ecosystem powered by Jenkins.
+
+Currently Pipeline is one of the best automation-as-code solutions available in automation servers, so I would like to highlight several Pipeline features, which may be interesting to Jenkins engineers working in the area.
+
+## Introduction
+
+In Embedded projects it's frequently required to run tests on specific hardware peripherals: development boards, prototypes, etc.
+It may be required for both software and hardware areas, and especially for products involving both worlds.
+CI and CD methodologies require continuous integration and system testing, and Jenkins comes to help here.
+Jenkins is an automation framework, which can be adjusted to reliably work with hardware attached to its nodes.
+
+The classic automation approaches in Jenkins are based on Free-style and Multi-configuration project types. 
+Links to various articles on this topic are collected on the 
+link:/solutions/embedded/[HW/Embedded Solution page]Embedded on the Jenkins website.
+Tests automation on hardware peripherals has been covered in several publications by Robert Martin, Steve Harris, JL Gray, Gordon McGregor, Martin d’Anjou, and Sarah Woodall.
+There is also a top-level overview of classic approaches made by me at JUC2015/London (a bit outdated now).
+
+On the other hand, there is no previous publications, which would address Automation-as-code usage for the Embedded area.
+In this post I want to address this use-case.
+
+## Area challenges
+
+Generally, any peripheral hardware device can be attached to a Jenkins node. 
+Since Jenkins nodes require Java only, almost every development machine can be attached.
+Below you can find a common connection scheme:
+
+image::/images/blog/pipeline-as-code-for-hardware/connectBoard.png["Connecting the external device", width=550, align="center"]
```

Comment:
Will you please move images into the `/content/images/post-images` directory. There's one imagine in that `blog/` directory that I need to move since I missed it in a code review.